sdks/java/nexmark/src/test/java/org/apache/beam/sdk/nexmark/model/sql/adapter/ModelFieldsAdaptersTest.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.sdk.nexmark.model.sql.adapter;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.List;
+
+import org.apache.beam.sdk.extensions.sql.BeamRecordSqlType;
+import org.apache.beam.sdk.nexmark.model.Auction;
+import org.apache.beam.sdk.nexmark.model.Bid;
+import org.apache.beam.sdk.nexmark.model.Person;
+
+import org.junit.Test;
+
+/**
+ * Unit tests for {@link ModelFieldsAdapters}.
+ */
+public class ModelFieldsAdaptersTest {
+
+  private static final Person PERSON =
+      new Person(3L, "name", "email", "cc", "city", "state", 329823L, "extra");
+
+  private static final BeamRecordSqlType PERSON_RECORD_TYPE = BeamRecordSqlTypeBuilder.of()
+      .withLongField("id")
```

Comment:
<!--thread_id:cc_151145780_t; commit:050bdab2afe544b8a5c58bfe1dfe1914a9820c8f; resolved:0-->
<!--section:context-quote-->
> **echauchot** wrote:
> Same comment as above: WDYT about doing this logic inside the model object to avoid spreading the knowledge of the fields of `Person` object outside of the `Person` class?

<!--section:body-->
I considered it but for the reasons in my other comment I decided against it.

This knowledge is essential for the SQL queries though. Queries need to know how to get data from records. These are just utility classes which determine this field mapping for specific set of SQL queries.